### PR TITLE
Add `CodePoints.toChars` functions

### LINKED ISSUE #9: Add `CodePoints.toChars` convenience
Equivalent of [`Character.toChars`](https://docs.oracle.com/en/java/javase/19/docs/api/java.base/java/lang/Character.html#toChars(int)).

### PR DESCRIPTION
**Pull request by JakeWharton**
Closes #9

### CODE REVIEW

File: src/commonImplementation/kotlin/CommonCodePoints.kt [hunk 1]
```diff
@@ -58,4 +58,26 @@ object CommonCodePoints {
...
+        } else {
+            charArrayOf(highSurrogate(codePoint), lowSurrogate(codePoint))
+        }
+    }
+
+    fun toChars(codePoint: Int, destination: CharArray, offset: Int): Int {
+        // We have to do an explicit bounds check first because of JS.
+        if (offset < 0 || offset >= destination.size - 1) {
+            throw IndexOutOfBoundsException("size: ${destination.size}, offset: $offset")
+        }
```
Comment (JakeWharton) [inline src/commonImplementation/kotlin/CommonCodePoints.kt]: I can pull this into an `expect`/`actual` that only actually runs on JS if you want. Every other platform has automatic bounds checks.

Comment (JakeWharton) [reply src/commonImplementation/kotlin/CommonCodePoints.kt]: Had a bug in the logic. It's harder to extract now but not impossible.

Comment (cketti) [reply src/commonImplementation/kotlin/CommonCodePoints.kt]: Let's leave it in for now. We can always optimize later should it become necessary.

**Comment by cketti** [comment]
Thanks :+1:

**Pull request merged**